api/pkg/s3/objectput.go

```diff
@@ -56,18 +56,22 @@ func (s *APIService) ObjectPut(request *restful.Request, response *restful.Respo
 
 	if isArchive == "Archive" && bucketMeta.Tiers != "" {
 		adminCtx := common.GetAdminContext()
-		backendId := s.GetBackendIdFromTier(adminCtx, bucketMeta.Tiers)
-		backendMeta, err := s.backendClient.GetBackend(adminCtx, &backend.GetBackendRequest{Id: backendId})
-		if err != nil {
-			log.Error("the selected backends from tier doesn't exists.")
-			response.WriteError(http.StatusInternalServerError, err)
+		backendMeta, backendErr := s.backendClient.ListBackend(adminCtx, &backend.ListBackendRequest{
```

Comment:
Where do we ensure that the backend has bucket?

Reply:
we are passing in filter, bucketMeta.defaultLocation

Reply:
got it